lib/charms/mongodb/v1/helpers.py

```diff
@@ -186,6 +188,24 @@ def get_mongod_args(
     full_conf_dir = f"{MONGODB_SNAP_DATA_DIR}{CONF_DIR}" if snap_install else CONF_DIR
     logging_options = _get_logging_options(snap_install)
     audit_log_settings = _get_audit_log_settings(snap_install)
+    if degraded:
+        cmd = [
+            # Only bind to local IP
+            f"--bind_ip {machine_ip}",
+            # db must be located within the snap common directory since the
+            # snap is strictly confined
+            f"--dbpath={full_data_dir}",
+            # for simplicity we run the mongod daemon on shards, configsvrs,
+            # and replicas on the same port
+            f"--port={Config.MONGODB_PORT}",
```

Comment:
Yes it does.
This is not something that can be done on sharded clusters because it would need some updates on both sides (config server and shard).
I thought I had specified this in the PR description but I might have forgotten and I'll add it :)